iOS/Flows/Conversation/ConversationListCoordinator+Presentation.swift

```diff
@@ -60,27 +60,41 @@ extension ConversationListCoordinator {
     }
 
     func presentPeoplePicker() {
+        // If there is no conversation to invite people to, then do nothing.
         guard let conversation = self.activeConversation else { return }
+
         let coordinator = PeopleCoordinator(router: self.router, deepLink: self.deepLink)
         coordinator.selectedConversationCID = self.activeConversation?.cid
-        
-        coordinator.toPresentable().dismissHandlers.append { [unowned self] in
-            if coordinator.invitedPeople.isEmpty {
-                self.presentDeleteConversationAlert(cid: coordinator.selectedConversationCID)
-            }
-        }
-        self.present(coordinator) { [unowned self] people in
-            
-            if people.isEmpty {
-                self.presentDeleteConversationAlert(cid: coordinator.selectedConversationCID)
-            } else {
-                self.add(people: people, to: conversation)
+
+        self.present(coordinator, finishedHandler: { [unowned self] invitedPeople in
+            self.handleInviteFlowEnded(givenInvitedPeople: invitedPeople, activeConversation: conversation)
+        }, cancelHandler: { [unowned self, unowned coordinator = coordinator] in
+            let invitedPeople = coordinator.invitedPeople
+            self.handleInviteFlowEnded(givenInvitedPeople: invitedPeople, activeConversation: conversation)
+        })
+    }
+
+    private func handleInviteFlowEnded(givenInvitedPeople invitedPeople: [Person],
+                                                       activeConversation: Conversation) {
+
+        if invitedPeople.isEmpty {
+            // If the user didn't invite anyone to the conversation and the conversation doesn't have
+            // any existing members, ask them if they'd like to delete it.
+            Task {
+                let peopleInConversation = await PeopleStore.shared.getPeople(for: activeConversation)
+                guard peopleInConversation.isEmpty else { return }
+
+                self.presentDeleteConversationAlert(cid: activeConversation.cid)
             }
+        } else {
+            // Add all of the invited people to the conversation.
+            self.add(people: invitedPeople, to: activeConversation)
```

Comment:
Improving the delete conversation logic. Delete shows if no one was invited AND there isn't anyone in the associated with the conversation (including contacts).